app/charts/area/areas-state.tsx

```diff
@@ -160,9 +160,13 @@ const useAreasState = (
     [dataGroupedByX, xKey, getY, getSegment]
   );
 
+  const plotters = useMemo(() => {
+    return [getX, getY];
+  }, [getX, getY]);
```

Comment:
It's a bit inconvenient to memo here, maybe usePlottableData could have the plotters directly as the dependency ?

in chart-helpers.tsx

```patch
  const isPlottable = useCallback(
    (d: Observation) => {
      for (let p of plotters) {
        const v = p(d);
        if (v === undefined || v === null) {
          return false;
        }
      }
      return true;
    },
-    [plotters]
+    plotters
  );
```

Reply:
Wow, that's a great hint, I totally missed that 😅 I'll adjust :)